Add .github/copilot-instructions.md for Copilot cloud agent onboarding (#4919)
cisagov/getgov: .github/copilot-instructions.md

# Copilot Instructions for manage.get.gov

## Project overview

> This repository contains the source code for the **.gov domain registrar** — the system used by U.S. government organizations to request and manage `.gov` domains. It is a Django 4.2 web application deployed on Cloud.gov (Cloud Foundry), backed by PostgreSQL, and integrated with Login.gov (OIDC) and an EPP-based domain registry.

Reviewer: ```suggestion
This repository contains the source code for the **.gov domain registrar** — the system used by U.S.-based government organizations to request and manage `.gov` domains. It is a Django 4.2 web application deployed on Cloud.gov (Cloud Foundry), backed by PostgreSQL, and integrated with Login.gov (OIDC) and an EPP-based domain registry.
```

### Templates (`src/registrar/templates/`)

Django templates using USWDS components. `base.html` and `dashboard_base.html` are the main layout templates. Custom accessible components follow USWDS patterns.

### Frontend assets (`src/registrar/assets/`)

- USWDS Sass files live in `registrar/assets/_theme/`.
- The `node` Docker service watches for changes and recompiles automatically via gulp.
- Custom images go in `registrar/assets/img/registrar/` and must be force-added to git: `git add --force <img-file>` (the `/img/` directory is in `.gitignore`).
- Compiled assets are served from `registrar/public/`.

### Feature flags

django-waffle is used for feature flags. Check and modify flags via Django Admin or management commands. The `WaffleFlag` model in `registrar/models/waffle_flag.py` mirrors waffle flags for admin display.

---

## Branch and PR conventions

- **Branch naming**: `initials/issue-number-description` (e.g., `ab/1234-update-documentation`). Branches starting with `initials/` auto-deploy to a personal sandbox when a PR exists.

> - **Merge strategy**: Always **merge**, never rebase. This project values merge context over linear history.

Reviewer 2: ```suggestion
- **Merge strategy**: Merge and squash to main.  Never rebase main. 
```
